src/heretic/mn_menu.c

```diff
@@ -627,7 +627,7 @@ void MN_DrTextA(const char *text, int x, int y)
 
     while ((c = *text++) != 0)
     {
-        if (c < 33)
+        if (c < 33 || c > 91) // [crispy] fail-safe: draw patches above FONTA59 as spaces
```

Comment:
This was a Right Thing to do in terms of fail-safe, but there is still some potential for improvement. Small example, I'm using British Antigua and Barbuda locale, which is using lowercase for `am/pm` designation. Unlike Doom, Heretic can't handle lowercase characters, that's why all game strings are always uppercased. In this case, designation can be drawed:

![image](https://github.com/fabiangreffrath/crispy-doom/assets/21193394/006310af-1531-4c1a-9cb3-384e49024389)

Possible solution is to forcefully convert lower case chars to uppercase, this will do the trick:
```diff
    while ((c = *text++) != 0)
    {
+++        if (c > 96 && c < 123) // means, all ASCII chars "a" ... "z"
+++            c -= 32;           // ... will be reindexed to "A" ... "Z"
```

But how to properly make this condition friendly with newly introduced `|| c > 91)` and avoid messy maze of conditions, that's a good question. 🤔

Reply:
Something like this, but needs support for big font (it doesn't have `[` symbol (№59) which is a cursor) and probably smarter conditions for indexes checking.

<details>

```diff
+++ international-doom/src/heretic/mn_menu.c	Wed Apr 24 08:41:20 2024
@@ -4148,6 +4148,24 @@
 //
 //---------------------------------------------------------------------------
 
+static const char MN_CheckValidChar (char ascii_index, int small_font)
+{
+    // Following characters not exising in font graphics, replace with spaces:
+    // \ ] ^ _ { | } ~
+    if ((ascii_index >= 92 && ascii_index <= 96) || ascii_index >= 123)
+    {
+        return 32;
+    }
+    // Force lowercase a...z characters to uppercase A...Z.
+    if (ascii_index >= 97 && ascii_index <= 122)
+    {
+        return ascii_index -= 32;
+    }
+
+    // Valid char, do not modify ASCII index.
+    return ascii_index;
+}
+
 void MN_DrTextA (const char *text, int x, int y, byte *table)
 {
     char c;
@@ -4157,6 +4175,8 void MN_DrTextA (const char *text, int x, int y, byte *table) @@
 
     while ((c = *text++) != 0)
     {
+        c = MN_CheckValidChar(c, 1);
+
         if (c < 33)
         {
             x += 5;
```

![image](https://github.com/fabiangreffrath/crispy-doom/assets/21193394/b85ff5cc-c545-48b1-b209-07e9cbc3e771)

</details>

This way, even such messy string will work: `E1M1:  THE  \\ {|} ~ dOcKs`